### PR TITLE
Marketing: re-enable SEO upsell and update its copy

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Re-enable SEO upsell from Marketing > Traffic section and update its copy.

#### Implementation notes

* Copy for non-Jetpack sites remains untouched.
* Clicking the upsell (Jetpack site only) takes users to Jetpack Security Options page.

#### Testing instructions

- Run this PR.
- Select a site with Jetpack Free.
- Visit /marketing/traffic/:site.
- Verify that you see an upsell (see capture below) in "Search engine optimization" panel.
- Verify that the upsell's copy look like the one in the capture.
- Click the upsell.
- Verify that you are redirected to Jetpack Security Options page (you should be able to see both Daily and Real-time option cards).
- Do the same with a site that has Jetpack Complete.
- Verify that you don't see the upsell.

Fixes 1196341175636977-as-1197426524966358

#### Demo
image

### CODE REVIEW

File: client/my-sites/site-settings/seo-settings/form.jsx [hunk 1]
```diff
@@ -353,11 +350,10 @@ export class SeoForm extends React.Component {
 
 				{ ! this.props.hasSeoPreviewFeature &&
 					! this.props.hasAdvancedSEOFeature &&
-					selectedSite.plan &&
-					! shouldShowOfferResetFlow() && (
+					selectedSite.plan && (
 						<UpsellNudge
 							description={ translate(
-								'Get tools to optimize your site for improved performance in search engine results.'
+								'Get tools to optimize your site for improved search engine results.'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/seo-settings/form.jsx:356]: :warning: This change will be queued for retranslation. We'll display the English text until then.

**Review by elliottprogrammer — APPROVED** [review]
LGTM! 👍

**Comment by rcanepa** [comment]
I initially forgot to update the unit tests that cover this section of the app so I pushed a new update to fix that.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4866366

Thank you @rcanepa for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4866366).

**Pull request merged**